apis/applicationintegration/v1alpha1/sqs_types.go

```diff
@@ -0,0 +1,169 @@
+	AttributeLastModifiedTimestamp                 string = "LastModifiedTimestamp"
+	AttributeQueueArn                              string = "QueueArn"
+	AttributeApproximateNumberOfMessagesDelayed    string = "ApproximateNumberOfMessagesDelayed"
+	AttributeDelaySeconds                          string = "DelaySeconds"
+	AttributeReceiveMessageWaitTimeSeconds         string = "ReceiveMessageWaitTimeSeconds"
+	AttributeRedrivePolicy                         string = "RedrivePolicy"
+	AttributeDeadLetterQueueARN                    string = "DeadLetterQueueARN"
+	AttributeMaxReceiveCount                       string = "MaxReceiveCount"
+	AttributeFifoQueue                             string = "FifoQueue"
+	AttributeContentBasedDeduplication             string = "ContentBasedDeduplication"
+	AttributeKmsMasterKeyID                        string = "KmsMasterKeyId"
+	AttributeKmsDataKeyReusePeriodSeconds          string = "KmsDataKeyReusePeriodSeconds"
+)
+
+// Tag is a key value pairs attached to a Amazon SQS queue.
+type Tag struct {
+
+	// The key name that can be used to look up or retrieve the associated value.
+	Key string `json:"key"`
+
+	// The value associated with a key in a tag.
+	Value string `json:"value,omitempty"`
+}
+
+// RedrivePolicy includes the parameters for the dead-letter queue functionality of the source queue.
+type RedrivePolicy struct {
+	// The Amazon Resource Name (ARN) of the dead-letter queue to which Amazon
+	// SQS moves messages after the value of maxReceiveCount is exceeded.
+	// +optional
+	DeadLetterQueueARN *string `json:"deadLetterQueueARN,omitempty"`
+
+	// The number of times a message is delivered to the source queue before
+	// being moved to the dead-letter queue.
+	// +optional
+	MaxReceiveCount *int64 `json:"maxReceiveCount,omitempty"`
+}
+
+// QueueParameters define the desired state of an AWS Queue
+type QueueParameters struct {
+	// The length of time, in seconds, for which the delivery
+	// of all messages in the queue is delayed.
+	// +optional
+	DelaySeconds *int64 `json:"delaySeconds,omitempty"`
+
+	// Designates a queue as FIFO.
+	// +immutable
+	// +optional
+	FifoQueue *bool `json:"fifoQueue,omitempty"`
+
+	// The limit of how many bytes a message can contain before Amazon SQS rejects it.
+	// +optional
+	MaximumMessageSize *int64 `json:"maximumMessageSize,omitempty"`
+
+	// The length of time, in seconds, for which Amazon SQS retains a message.
+	// +optional
+	MessageRetentionPeriod *int64 `json:"messageRetentionPeriod,omitempty"`
+
+	// The name of the new queue.
+	Name string `json:"name"`
+
+	// The length of time, in seconds, for which a ReceiveMessage
+	// action waits for a message to arrive.
+	// +optional
+	ReceiveMessageWaitTimeSeconds *int64 `json:"receiveMessageWaitTimeSeconds,omitempty"`
+
+	// RedrivePolicy includes the parameters for the dead-letter queue
+	// functionality of the source queue.
+	// +optional
+	RedrivePolicy RedrivePolicy `json:"redrivePolicy,omitempty"`
```

Comment:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

I believe this should be a pointer.